imx/deposit.go

```diff
@@ -59,17 +59,41 @@ func NewERC721Deposit(tokenID, tokenAddress string) *ERC721Deposit {
 	return &this
 }
 
-func (c *Client) getSignableDeposit(
-	ctx context.Context,
-	request *api.GetSignableDepositRequest) (*api.GetSignableDepositResponse, error) {
-	signableDepositResponse, httpResp, err := c.GetSignableDeposit(ctx).GetSignableDepositRequest(*request).Execute()
+/*
+GetDeposit Gets details of a deposit with the given ID
+
+@param ctx context.Context - for cancellation, deadlines, tracing, etc or context.Background().
+@param id Deposit ID
+@return Deposit
+*/
+func (c *Client) GetDeposit(ctx context.Context, id string) (*api.Deposit, error) {
+	response, httpResponse, err := c.depositsApi.GetDeposit(ctx, id).Execute()
 	if err != nil {
-		return nil, fmt.Errorf("error when calling `Deposits.GetSignableDeposit`: %v, HTTP response body: %v", err, httpResp.Body)
+		return nil, fmt.Errorf("error in getting the deposit details: %v, HTTP response body: %v", err, httpResponse.Body)
 	}
-	return signableDepositResponse, nil
+	return response, nil
+}
+
+/*
+ListDeposits Gets a list of deposits
+
+@param ctx context.Context - for cancellation, deadlines, tracing, etc or context.Background().
+@return ListDepositsResponse
+*/
+func (c *Client) ListDeposits(ctx context.Context) (*api.ListDepositsResponse, error) {
+	response, httpResponse, err := c.depositsApi.ListDeposits(ctx).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("error in getting the deposits list: %v, HTTP response body: %v", err, httpResponse.Body)
+	}
+	return response, nil
 }
 
-// Deposit performs the deposit workflow on the ETHDeposit.
+/*
+Deposit performs the deposit workflow on the ETHDeposit.
+
+@param ctx context.Context - for cancellation, deadlines, tracing, etc or context.Background().
```

Comment:
might be nice to add details about the other params in the method signature as well.

Reply:
Love the params docs by the way.

Reply:
> might be nice to add details about the other params in the method signature as well.
I can't see other params other than the context and the return object...
The receiver `*Client` is omitted because it's the struct itself, which (usually) has a separate docstring